src/app/core/campaign/campaign.service.ts

```diff
@@ -14,7 +14,15 @@ export class CampaignService {
 
   getCampaign(id: number | string): Observable<CampaignState> {
     return this.augury.follow('prx:campaign', { id, zoom: 'prx:flights' }).pipe(
-      switchMap(doc => doc.followItems('prx:flights').pipe(map(flightDocs => ({ doc, flightDocs })))),
+      switchMap(doc => doc.follow('prx:flights').pipe(map(flights => ({ doc, flights })))),
+      switchMap(({ doc, flights }: { doc: HalDoc; flights: HalDoc }) => {
+        // if total is greater than count, request all flights
+        let params: any;
+        if (+flights['total'] > +flights['count']) {
+          params = { per: +flights['total'] };
+        }
+        return doc.followItems('prx:flights', params).pipe(map(flightDocs => ({ doc, flightDocs })));
```

Comment:
So this should work whether or not `prx:flights` comes back paged or not, right?  (This would be a good addition to HalObservable itself, someday).